### ISSUE TITLE
Unable to retrieve turn based matches using GetAllMatches

### ISSUE THREAD

**Issue by jamstone** [playgameservices/play-games-plugin-for-unity#981]
public void GetPendingMatches()
        {
            Debug.Log("Before GetAllMatches is called.");
            PlayGamesPlatform.Instance.TurnBased.GetAllMatches(GetAllThoseMatches);
        }

        public void GetAllThoseMatches(TurnBasedMatch[] matches)
        {
            string _mStatus = "Got " + matches.Length + " matches";
            foreach (TurnBasedMatch match in matches)
            {
                _mStatus += " " + match.MatchId + " " +
                    match.Status + " " +
                    match.TurnStatus + "\n";
            }
            Debug.Log(_mStatus);
        }

The console only shows "Before GetAllMatches is called." when I call GetPendingMatches. I am still able to view matches through the inbox. I have also tried copy-pasting the implementation in SmokeTest to no avail.

Using version 9.30 with Unity 5.3.1.

Upon further digging in the API...

In NativeTurnBasedMultiplayerClient:

        public void GetAllMatches(Action<TurnBasedMatch[]> callback)
        {
            UnityEngine.Debug.Log("Ping!");

            mTurnBasedManager.GetAllTurnbasedMatches((allMatches) =>
                {

                    UnityEngine.Debug.Log("Pong!");

                    int count = allMatches.MyTurnMatchesCount() +
                        allMatches.TheirTurnMatchesCount() +
                        allMatches.CompletedMatchesCount();

                    TurnBasedMatch[] matches = new TurnBasedMatch[count];
                    int i=0;
                    foreach (var match in allMatches.MyTurnMatches())
                    {
                        matches[i++] = match.AsTurnBasedMatch(mNativeClient.GetUserId());
                    }
                    foreach (var match in allMatches.TheirTurnMatches())
                    {
                        matches[i++] = match.AsTurnBasedMatch(mNativeClient.GetUserId());
                    }
                    foreach (var match in allMatches.CompletedMatches())
                    {
                        matches[i++] = match.AsTurnBasedMatch(mNativeClient.GetUserId());
                    }
                    callback(matches);
                });
        }

I am getting "Ping!" but not "Pong!".

In TurnBasedManager:

        internal void GetAllTurnbasedMatches(Action<TurnBasedMatchesResponse> callback)
        {
            UnityEngine.Debug.Log("Beep!");

            Action<TurnBasedMatchesResponse> myCallback = cb =>
            {
                UnityEngine.Debug.Log("Boop!");

                callback(cb);
            };

            TurnBasedMultiplayerManager.TurnBasedMultiplayerManager_FetchMatches(mGameServices.AsHandle(),
                InternalTurnBasedMatchesCallback,
                Callbacks.ToIntPtr<TurnBasedMatchesResponse>(
                    myCallback, TurnBasedMatchesResponse.FromPointer));
        }

I am getting "Beep!" but not "Boop!".

And beyond this seems to be a DLL import in TurnBasedMultiplayerManager that I have no idea how to troubleshoot.

Basically it seems like the callback argument is being ignored in whatever method is being imported from that DLL.

**Comment by claywilkinson** [comment]
Hmm. so I think Debug.Log only works on the UI thread.  I'll make a note to update these sometime.  Thanks for the followup.
